server/app/assets/javascripts/admin_predicate_configuration.ts

```diff
@@ -0,0 +1,530 @@
+import {addEventListenerToElements, assertNotNull} from './util'
+
+/** Dynamic behavior for ProgramBlockPredicateConfigureView.
+ *
+ * When scalar is selected, updates operator and value input(s) to match.
+ * When operator is selected, updates value input(s) to match.
+ * Implements adding and deleting value rows when the user clicks on the UI.
+ * */
+class AdminPredicateConfiguration {
+  registerEventListeners() {
+    addEventListenerToElements('.cf-scalar-select', 'input', (event: Event) =>
+      this.configurePredicateFormOnScalarChange(event),
+    )
+
+    addEventListenerToElements('.cf-operator-select', 'input', (event: Event) =>
+      this.configurePredicateFormOnOperatorChange(event),
+    )
+
+    // Trigger a select event to set the correct input type on the value field(s)
+    Array.from(document.querySelectorAll('.cf-operator-select select')).forEach(
+      (el) => {
+        const event = new CustomEvent('input', {bubbles: true})
+        el.dispatchEvent(event)
+      },
+    )
+
+    document
+      .querySelector('#predicate-add-value-set')
+      ?.addEventListener('click', (event: Event) =>
+        this.predicateAddValueRow(event),
+      )
+
+    addEventListenerToElements(
+      '.predicate-config-delete-value-row',
+      'click',
+      (event: Event) => this.predicateDeleteValueRow(event),
+    )
+  }
+
+  configurePredicateFormOnScalarChange(event: Event) {
+    // Get the type of scalar currently selected.
+    const scalarDropdown = event.target as HTMLSelectElement
+    const selectedScalarType =
+      scalarDropdown.options[scalarDropdown.options.selectedIndex].dataset.type!
+    const selectedScalarValue =
+      scalarDropdown.options[scalarDropdown.options.selectedIndex].value
+
+    this.filterOperators(
+      scalarDropdown,
+      selectedScalarType,
+      selectedScalarValue,
+    )
+  }
+
+  configurePredicateFormOnOperatorChange(event: Event) {
+    const operatorDropdown = event.target as HTMLSelectElement
+    const selectedOperatorValue =
+      operatorDropdown.options[operatorDropdown.options.selectedIndex].value
+    const operatorDropdownContainer = assertNotNull(
+      operatorDropdown.closest('.cf-operator-select'),
+    ) as HTMLElement
+
+    const questionId = assertNotNull(
+      operatorDropdownContainer.dataset.questionId,
+    )
+
+    const commaSeparatedHelpTexts = document.querySelectorAll(
```

Comment:
A section doc would be nice here.  What are the Help texts and where are we showing them and in what situation?

Reply:
done

Reply:
done